### ISSUE TITLE
Synapse 1.86 can't start anymore with existing SQLite database

### ISSUE THREAD

**Issue by patois87** [matrix-org/synapse#15837]
### Description

After updating to Synapse 1.86 the server can't start anymore under Debian 12. I use SQLite as database.

### Steps to reproduce

- I tried to use a backup from my database, which I know to be working, but the error peresists.
- If I roll back to Synapse 1.85.2 the server is working again.
- Also when I delete the database and start Synpase 1.86, the server creates a new database, which then is working.

### Homeserver

personal homeserver

### Synapse Version

1.86.0

### Installation Method

Debian packages from packages.matrix.org

### Database

SQLite

### Workers

Single process

### Platform

Debian 12

### Relevant log output

```shell
Jun 26 10:59:16 server matrix-synapse[818479]: **********************************************************************************
Jun 26 10:59:16 server matrix-synapse[818479]:  There may be more information in the logs.
Jun 26 10:59:16 server matrix-synapse[818479]:  
Jun 26 10:59:16 server matrix-synapse[818479]:      sqlite3.IntegrityError: UNIQUE constraint failed: temp_user_filters.user_id
Jun 26 10:59:16 server matrix-synapse[818479]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
Jun 26 10:59:16 server matrix-synapse[818479]:          return func(sql, *args, **kwargs)
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/database.py", line 469, in _do_execute
Jun 26 10:59:16 server matrix-synapse[818479]:          self._do_execute(self.txn.execute, sql, parameters)
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/database.py", line 417, in execute
Jun 26 10:59:16 server matrix-synapse[818479]:          cur.execute(copy_sql, (f"{hostname}",))
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/schema/main/delta/78/02_validate_and_update_user_filters.py", line 85, in run_upgrade
Jun 26 10:59:16 server matrix-synapse[818479]:          module.run_upgrade(cur, database_engine, config=config)
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/prepare_database.py", line 514, in _upgrade_existing_database
Jun 26 10:59:16 server matrix-synapse[818479]:          _upgrade_existing_database(
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/prepare_database.py", line 141, in prepare_database
Jun 26 10:59:16 server matrix-synapse[818479]:          prepare_database(
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/storage/databases/__init__.py", line 74, in __init__
Jun 26 10:59:16 server matrix-synapse[818479]:                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jun 26 10:59:16 server matrix-synapse[818479]:          self.datastores = Databases(self.DATASTORE_CLASS, self)
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/server.py", line 339, in setup
Jun 26 10:59:16 server matrix-synapse[818479]:          hs.setup()
Jun 26 10:59:16 server matrix-synapse[818479]:        File "/opt/venvs/matrix-synapse/lib/python3.11/site-packages/synapse/app/homeserver.py", line 352, in setup
Jun 26 10:59:16 server matrix-synapse[818479]:      Traceback (most recent call last):
Jun 26 10:59:16 server matrix-synapse[818479]:  Error during initialisation:
Jun 26 10:59:16 server matrix-synapse[818479]: **********************************************************************************
```

**Comment by toymil** [comment]
@patois87 did you solve it?  I am having the same issue and error message.

**Comment by alariej** [comment]
Hi @patois87, I'm getting the exact same issue using the docker image. Can't upgrade past 1.86 :-( Can you give me a hint on how you fixed this? Thanks.

**Comment by patois87** [comment]
Hi @alariej, I use the Debian package and so I was able to manually install the previous version.  The matrix team released version 1.87 yesterday, which should fix the problem.

**Comment by alariej** [comment]
Excellent, I just updated to 1.87 and everything is working :-) Thanks!
